keps/sig-cli/4292-kubectl-debug-custom-profile/README.md

```diff
@@ -0,0 +1,981 @@
+`export KUBECTL_DEBUG_CUSTOM_PROFILE=true`. User can easily disable this feature
+via unsetting this environment variable.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [ ] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name:
+  - Components depending on the feature gate:
+- [X] Other
+  - Describe the mechanism: export KUBECTL_DEBUG_CUSTOM_PROFILE=true
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+Enabling `KUBECTL_DEBUG_CUSTOM_PROFILE` environment variable will enable a new `custom` flag
+in `kubectl debug`.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+Yes, the way of enabling this feature is exporting `KUBECTL_DEBUG_CUSTOM_PROFILE=true`.
+The way of disabling is simply unsetting this environment.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+Flag becomes visible and no risk for the cluster.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+In alpha phase, all the tests can be done only after enabling this feature.
```

Comment:
This answer sort of reads like it missed the question (?), but this feature doesn't change behavior of stored objects anyhow.

Reply:
Yes, updated this answer to more align with the question. Please let me know what you think.